osu.Framework/Graphics/UserInterface/TextBox.cs

```diff
@@ -832,6 +842,49 @@ protected virtual void OnCaretMoved(bool selecting)
         {
         }
 
+        /// <summary>
+        /// Invoked whenever the selected text has changed.
+        /// </summary>
+        /// <param name="selectionType">The type of selection change that occured.</param>
+        protected virtual void OnTextSelectionChanged(TextSelectionType selectionType)
+        {
+        }
+
+        /// <summary>
+        /// Invoked whenever text is deselected.
+        /// </summary>
+        protected virtual void OnTextDeselected()
+        {
+        }
+
+        private void onTextSelectionChanged(TextSelectionType selectionType)
+        {
+            if (lastSelectionStart == selectionStart && lastSelectionEnd == selectionEnd) return;
+
+            switch (selectionType)
+            {
+                case TextSelectionType.Deselect:
+                    if (lastSelectionStart != lastSelectionEnd && (lastSelectionStart > 0 || lastSelectionEnd > 0))
```

Comment:
```suggestion
                    if (lastSelectionStart != lastSelectionEnd)
```

| `lastSelectionStart` | `lastSelectionEnd` | `lastSelectionStart != lastSelectionEnd` | `lastSelectionStart > 0 \|\| lastSelectionEnd > 0` |
| -- | -- | -- | -- |
| 0 | 0 | false | false |
| 0 | 5 | true | true |
| 4 | 0 | true | true |
| 4 | 5 | true | true |
| 4 | 4 | false | true |